src/prpy/util.py

```diff
@@ -749,3 +749,178 @@ def IsTimedTrajectory(trajectory):
     """
     cspec = trajectory.GetConfigurationSpecification()
     return cspec.ExtractDeltaTime(trajectory.GetWaypoint(0)) is not None
+
+
+def JointStatesFromTraj(robot, traj, times, derivatives=[0, 1, 2]):
+    """
+    Helper function to extract the joint position, velocity and acceleration
+    from an OpenRAVE trajectory.
+    @param robot The OpenRAVE robot
+    @param traj An OpenRAVE trajectory
+    @param times List of times in seconds
+    @param derivatives list of desired derivatives (only supports 0, 1, 2),
+                       defaults to [0, 1, 2]
+    @return pva_list List of list of position, velocity and acceleration
+                     at specified times.
+                     Inserts 'None' for unavailable or undesired fields
+                     The i-th element is the i-th derivative of position
+                     Of size |times| x 3
+
+    """
+    for d in derivatives:
+        if not (d == 0 or d == 1 or d == 2):
+            raise ValueError('Derivative input {0:} is not 0, 1, or 2'
+                             .format(derivatives))
+
+    duration = traj.GetDuration()
+
+    if any(times > duration):
+        raise ValueError('Input times {0:} exceed duration {1:.2f}'
+                         .format(times, duration))
+
+    cspec = traj.GetConfigurationSpecification()
+    num_dofs = robot.GetDOF()
+    dof_indices = range(num_dofs)
+
+    pva_list = []
+    for t in times:
+        pva = [None] * 3
+        trajdata = traj.Sample(t)
+        for d in derivatives:
+            pva[d] = cspec.ExtractJointValues(trajdata, robot,
+                                              dof_indices, d)
```

Comment:
I find arbitrary derivative ordering a bit dangerous since there's no way to recover the derivative semantic from the `JointStates` list. I would actually consider doing something like:
```
for i in range(3):
    if i in derivatives:
        pva[i] = cspec.ExtractJointValues(trajdata, robot, dof_indices, i)
```